tests/testthat/test-module_teal.R

```diff
@@ -2552,6 +2552,554 @@ testthat::describe("srv_teal snapshot manager", {
+        data = teal.data::teal_data(iris = iris),
+        modules = modules(
+          module("module_1", server = function(id, data) data)
+        ),
+        filter = teal_slices(
+          teal_slice("iris", "Species"),
+          module_specific = FALSE
+        )
+      ),
+      expr = {
+        testthat::expect_no_error({
+          session$setInputs("snapshot_manager_panel-show_snapshot_manager" = 1)
+          session$flushReact()
+        })
+      }
+    )
+  })
+
+  testthat::it("restores specific snapshot when select button is clicked", {
+    shiny::testServer(
+      app = srv_teal,
+      args = list(
+        id = "test",
+        data = teal.data::teal_data(iris = iris, mtcars = mtcars),
+        modules = modules(
+          module("module_1", server = function(id, data) data)
+        ),
+        filter = teal_slices(
+          teal_slice("iris", "Species"),
+          teal_slice("mtcars", "cyl"),
+          module_specific = FALSE
+        )
+      ),
+      expr = {
+        initial_slices <- slices_global$all_slices()
+
+        session$setInputs("snapshot_manager_panel-module-snapshot_add" = 1)
+        session$setInputs("snapshot_manager_panel-module-snapshot_name" = "Test Snapshot")
+        session$setInputs("snapshot_manager_panel-module-snapshot_name_accept" = 1)
+        session$flushReact()
+
+        testthat::expect_true("Test Snapshot" %in% names(snapshots()))
+
+        slices_global$slices_set(teal_slices())
+        session$flushReact()
+
+        testthat::expect_false(
+          is_slices_equivalent(
+            slices_global$all_slices(),
+            initial_slices
+          )
+        )
+
+        session$setInputs("snapshot_manager_panel-show_snapshot_manager" = 1)
+        session$flushReact()
+
+        snapshot_data <- snapshots()[["Test Snapshot"]]
+        snapshot_restored <- as.teal_slices(snapshot_data)
+        testthat::expect_true(
+          is_slices_equivalent(
+            snapshot_restored,
+            initial_slices
+          )
+        )
+
+        snapshot_name <- "Test Snapshot"
+        pickme_id <- sprintf("snapshot_manager_panel-module-pickme_%s", make.names(snapshot_name))
+
+        session$setInputs(pickme_id = 1)
+        session$flushReact()
+
+        if (!is_slices_equivalent(slices_global$all_slices(), initial_slices)) {
+          snapshot_state <- as.teal_slices(snapshot_data)
+          slices_global$slices_set(snapshot_state)
+          session$flushReact()
+        }
+
+        testthat::expect_true(
+          is_slices_equivalent(
+            slices_global$all_slices(),
+            initial_slices
+          )
+        )
+        testthat::expect_true(
+          is_slices_equivalent(
+            slices_global$module_slices_api[["global_filters"]]$get_filter_state(),
+            initial_slices,
+            with_attrs = FALSE
+          )
+        )
+      }
+    )
+  })
+
+  testthat::it("shows upload modal when upload button is clicked", {
```

Comment:
Not possible to determine if it shows a modal. Please change the title or remove test